protos/sigstore_common.proto

```diff
@@ -0,0 +1,105 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.common.v1;
+
+option go_package = "github.com/sigstore/proto/common/v1";
+option java_package = "dev.sigstore.proto.common.v1";
+option java_multiple_files = true;
+option java_outer_classname = "CommonProto";
+
+// This package defines commonly used message types within the Sigstore
+// community.
+
+// Only a subset of the secure hash standard algorithms are supported.
+// See https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf for more
+// details.
+// UNSPECIFIED SHOULD not be used, primary reason for inclusion is to force
+// any proto JSON serialization to emit the used hash algorithm, as default
+// option is to *omit* the default value of an emum (which is the first
+// value, represented by '0'.
+enum HashAlgorithm {
+        UNSPECIFIED = 0;
+        SHA2_256 = 1;
+        SHA2_512 = 2;
+}
+
+// HashOutput captures a digest of a 'message' (generic octet sequence)
+// and the corresponding hash algorithm used.
+message HashOutput {
+        HashAlgorithm algorithm = 1;
+        // This is the raw octets of the message digest as computed by
+        // the hash algorithm.
+        bytes digest = 2;
+}
+
+// MessageSignature stores the computed signature over a message.
+message MessageSignature {
+        // Message digest can be used to identify the artifact.
+        HashOutput message_digest = 1;
+        // The raw bytes as returned from the signature algorithm.
+        // The signature algorithm (and so the format of the signature bytes)
+        // are determined by the contents of the 'verification_material',
+        // either a key-pair or a certificate. If using a certificate, the
+        // certificate contains the required information on the signature
+        // algorithm.
+        // When using a key pair, the algorithm MUST be part of the public
+        // key, which MUST be communicated out-of-band.
+        bytes signature = 2;
+}
+
+// This message holds a RFC 3161 timestamp.
+message RFC3161SignedTimestamp {
+        // Signed timestamp is the DER encoded TimeStampToken
+        // See https://www.rfc-editor.org/rfc/rfc3161.html#section-2.4.2
+        // For verification, the TimeStampToken MUST contain TSA's
+        // certificate chain.
+        bytes signed_timestamp = 1;
+}
+
+
+// PublicKeyIdentifier can be used to identify an (out of band) delivered
+// key, to verify a signature.
+message PublicKeyIdentifier {
+        // Optional unauthenticated hint on which key to use.
+        // The format of the hint must be agreed upon out of band by the
+        // signer and the verifiers, and so is not subject to this
+        // specification.
+        string hint = 1;
```

Comment:
Sure, one example I can easily think of: the hex encoded hash of the public key, which is quite common for key ids.

Reply:
Do we want to add that as an example, how that hint might be encoded? 

I also wonder, do we want to be more prescriptive about what hints are accepted, rather than any unauthenticated string?

Reply:
Yes, I was thinking about this this morning, will update this during the day.

Reply:
There's some guidance on verious RFCs. The one guidance would be to derive it from the public key value or have another means of ensuring uniqueness.
> The value of the keyIdentifier field SHOULD be derived from the
   public key used to verify the certificate's signature or a method
   that generates unique values.  

https://www.rfc-editor.org/rfc/rfc3280#section-4.2.1.1

Reply:
Updated.